packages/uploadthing/src/sdk/index.ts

```diff
@@ -30,105 +34,88 @@ import type {
   UrlWithOverrides,
   UTApiOptions,
 } from "./types";
+import { UTFile } from "./ut-file";
 import {
-  getApiKeyOrThrow,
+  downloadFiles,
   guardServerOnly,
   parseTimeToSeconds,
   uploadFilesInternal,
 } from "./utils";
 
-interface UTFilePropertyBag extends BlobPropertyBag {
-  lastModified?: number;
-  customId?: string;
-}
-
-/**
- * Extension of the Blob class that simplifies setting the `name` and `customId` properties,
- * similar to the built-in File class from Node > 20.
- */
-export class UTFile extends Blob {
-  name: string;
-  lastModified: number;
-  customId?: string;
-
-  constructor(parts: BlobPart[], name: string, options?: UTFilePropertyBag) {
-    const optionsWithDefaults = {
-      ...options,
-      type: options?.type ?? (lookup(name) || undefined),
-      lastModified: options?.lastModified ?? Date.now(),
-    };
-    super(parts, optionsWithDefaults);
-    this.name = name;
-    this.customId = optionsWithDefaults.customId;
-    this.lastModified = optionsWithDefaults.lastModified;
-  }
-}
+export { UTFile };
 
 export class UTApi {
   private fetch: FetchEsque;
-  private apiKey: string | undefined;
-  private defaultHeaders: Record<string, string>;
+  private defaultHeaders: FetchContextTag["baseHeaders"];
   private defaultKeyType: "fileKey" | "customId";
 
   constructor(opts?: UTApiOptions) {
+    const apiKey = getApiKeyOrThrow(opts?.apiKey);
+
+    // Assert some stuff
+    guardServerOnly();
```

Comment:
Should these be the first thing, even before getting api key?

Reply:
sure - may be helpful if the api key is undefined due to not being on the server